addons/account/wizard/base_document_layout.py

```diff
@@ -1,13 +1,70 @@
+    vat = fields.Char(related='company_id.vat', readonly=False,)
+    acc_number = fields.Char(compute='_compute_acc_number', inverse='_inverse_acc_number',)
+
     def document_layout_save(self):
         """Save layout and onboarding step progress, return super() result"""
         res = super(BaseDocumentLayout, self).document_layout_save()
         if step := self.env.ref('account.onboarding_onboarding_step_base_document_layout', raise_if_not_found=False):
             for company_id in self.company_id:
                 step.with_company(company_id).action_set_just_done()
+            if res.get('context') is not None:
+                res['context']['dialog_size'] = 'large'
         return res
+
+    @api.depends('report_layout_id')
+    def _compute_from_invoice(self):
+        move = self.env['account.move'].browse(self.env.context.get('active_id'))
+        for record in self:
+            if self.env.context.get('active_model') == 'account.move':
+                record.from_invoice = move.move_type == 'out_invoice'
+            else:
+                record.from_invoice = False
+
+    @api.depends('qr_code')
+    def _compute_preview(self):
+        """
+            This override is needed to add the depends on the qr code and the move
+        """
+        ir_ui_view, wizard_with_logo, preview_css, is_html_empty = super()._compute_preview()
+        if self.env.context.get('active_model') == 'account.move':
+            move = self.env['account.move'].browse(self.env.context.get('active_id'))
+            # If there is a move in the context then, the value put in the preview come from the move
+            if move:
+                #We don't want to display the qr_code twice since we decided to put a fake one.
+                move.display_qr_code = False
+                self.preview = ir_ui_view._render_template('account.report_invoice_wizard_iframe', {
+                    'company': wizard_with_logo,
+                    'preview_css': preview_css,
+                    'is_html_empty': is_html_empty,
+                    'o': move,
+                    'qr_code': self.qr_code,
+                })
+
+    @api.depends('partner_id', 'acc_number')
+    def _compute_acc_number(self):
+        for record in self:
+            if record.partner_id.bank_ids:
+                record.acc_number = record.partner_id.bank_ids[0].acc_number or ''
+            else:
+                record.acc_number = ''
+
+    def _inverse_acc_number(self):
+        for record in self:
+            if record.qr_code:
+                if record.partner_id.bank_ids and record.acc_number:
+                    record.partner_id.bank_ids.update({'acc_number': record.acc_number})
+                elif record.acc_number:
+                    record.partner_id.bank_ids = [
+                        (0, 0, {
+                            'acc_number': record.acc_number,
+                            'partner_id': record.partner_id.id,
+                        })
```

Comment:
Why don't we use the ORM Command? 

```suggestion
                       Command.create({
                            'acc_number': record.acc_number,
                            'partner_id': record.partner_id.id,
                        })
```